yaml_edit/lib/src/editor.dart

```diff
@@ -576,8 +576,21 @@ class YamlEditor {
     ArgumentError.checkNotNull(path, 'path');
 
     final expectedTree = _deepModify(_contents, path, [], expectedNode);
+    final initialYaml = _yaml;
     _yaml = edit.apply(_yaml);
-    _initialize();
+    try {
+      _initialize();
+    } on YamlException {
+      throw AssertionError('''
+Failed to produce valid YAML after modification.
+
+Initial Yaml:
+$initialYaml
+
+Final Yaml:
+$_yaml
```

Comment:
```suggestion
Internal error in package:yaml_edit -- Failed to produce valid YAML after modification.

# Initial Yaml:
> ${initialYaml.replaceAll('\n', '\n> ')}

# Final Yaml:
> ${_yaml.replaceAll('\n', '\n> ')}

Please file an issue at:
https://github.com/google/dart-neats/issues/new?labels=pkg%3Ayaml_edit%2C+pending-triage&template=yaml_edit.md&title=
```

I propose we something similar for the other assertion error below, maybe we make a private method for creating an `AssertionError`, like `_createAssertionError(String message, String oldYaml, String newYaml)`